src/lib/Libifl/tm.c

```diff
@@ -630,12 +630,18 @@ tm_init(void *info, struct tm_roots *roots)
 		return TM_ESYSTEM;
 
 	pbs_tcp_interrupt = 1;
-	if ((tm_jobid = getenv("PBS_JOBID")) == NULL)
+	if ((env = getenv("PBS_JOBID")) == NULL)
 		return TM_EBADENVIRONMENT;
-	tm_jobid_len = strlen(tm_jobid);
-	if ((tm_jobcookie = getenv("PBS_JOBCOOKIE")) == NULL)
+	tm_jobid_len = strlen(env);
+	tm_jobid = (char *)calloc(1, tm_jobid_len + 1);
```

Comment:
Please check for return value of calloc and also you need to free these allocated memory in tm_finalize()